### PR TITLE
perf: stop overfetching of instrument data

### PR DESCRIPTION
**Pull request by TCMeldrum**
Closes: https://github.com/UserOfficeProject/issue-tracker/issues/1223

## Description
This PR optimizes the fetching of instrument data by introducing a minimal version of instrument data fetching. 

## Motivation and Context
This change is required to improve the performance of the application by minimizing the amount of data fetched when full instrument data is not necessary.

## Changes
1. Introduced a minimal version of instrument data fetching in various components where full instrument data was previously fetched.
2. Updated the types and usages of instrument data in various components to use the minimal version.
3. Replaced the calls to `useInstrumentsData()` with `useInstrumentsMinimalData()` in various components.

## Additional Context
This change will reduce the load on the server and improve the user experience by speeding up the loading times.

## Fixes Jira Issue

[https://jira.esss.lu.se/browse/](https://jira.esss.lu.se/browse/)

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: apps/frontend/src/hooks/instrument/useInstrumentsMinimalData.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import {
```
Comment (Scott-James-Hurley) [inline apps/frontend/src/hooks/instrument/useInstrumentsMinimalData.ts:1]: This looks very similar to `useInstrumentsData.ts`. Should duplicated code be reduced or would it be preferable to keep them entirely separate?

Comment (TCMeldrum) [reply apps/frontend/src/hooks/instrument/useInstrumentsMinimalData.ts:1]: I tried to, but I could not for the life of me get the typing to work in some way you would have something along the lines of `InstrumentDataMinimal | InstrumentData` which I could not get to type check in the places that needed `InstrumentData`. Happy to be proven wrong, but I could not make it work when I tryed.

**Comment by TCMeldrum** [comment]
> This is grand. If optimisation is going to be an ongoing thing across teams, should best practice and design be discussed and agreed upon in order to make sure this effort is sustainable and doesn't go to waste down the road?

Maybe I can bring it up in the performance project meeting

File: apps/frontend/src/hooks/instrument/useInstrumentsMinimalData.ts (hunk 1 above)
Comment (Scott-James-Hurley) [reply apps/frontend/src/hooks/instrument/useInstrumentsMinimalData.ts:1]: image
I'll take that pay raise now, please.

**Review by Scott-James-Hurley — APPROVED** [review]

**Review by janosbabik — APPROVED** [review]
Hi @TCMeldrum, It looks good to me. However, do you have any evaluation data from before and after the changes? Having some statistics would be helpful when making performance-related changes.

**Comment by TCMeldrum** [comment]
These numbers are against my local so take with a grain of salt.
We have about 40 instruments each with about 5-10 instrument scis attached, so is a comparison between these 2 calls:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/ofg87678/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/ofg87678/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">

Minimal | Full
-- | --
0.039487s | 1.592049s
0.010098s | 1.487671s
0.013802s | 1.058054s
0.008792s | 1.163378s
0.008592s | 1.495123s
0.006798s | 1.543326s



</body>

</html>


Keep in mind this query was uses across multiple pages including the landing pages for all roles bar `user`.

**Pull request merged**